### ISSUE TITLE
Cannot create companion GCC Cross-toolchain without default platforms

### ISSUE THREAD

**Issue by JacisNonsense** [gradle/gradle#2909]
### Expected Behavior

Toolchains (specifically GCC) should not automatically inherit all platforms ability to build. 

```gradle
model {
 platforms {
    x86 { architecture 'x86' }
    x64 { architecture 'x86_64' }
    xaarch64 { architecture 'aarch64' } // Cross-platform
 }

 toolchains {
    gcc(Gcc) {
       target("x86") { }
       target("x64") { }
    }
    xgcc(Gcc) {      // Cross platform toolchain
       target("xaarch64") { cppCompiler = "aarch64-linux-gnueabi-g++" }
    }
 }
}
```
The `gcc` toolchain should only support x86 and x64 targets, while the `xgcc` toolchain should only support xaarch64 targets.

### Current Behavior

Both `gcc` and `xgcc` bind to all platform targets (x64, x86, xaarch64), as shown by `println <toolchain>.platforms`

### Context

The current behaviour makes it particularly difficult to create projects that can build for multiple targets (local platforms, as well as cross-platforms for remote targets). When using the following model rule, every binary reports buildable by xgcc (for a component that uses any targetplatform x86, x64 and/or xaarch64): 
```gradle
@Mutate
void configureOptionalBuildables(BinaryContainer binaries) {
    binaries.withType(NativeBinarySpec) { bin ->
       println "${bin} ${bin.toolChain}"
    }
 }
```

For my specific use case, I'd like to skip over builds that aren't supported on the current system (i.e. if a cross toolchain is supplied, but doesn't exist on the system path, just skip it, similar to how specifying an "arm" platform (with no custom prefix or path) will be skipped in favor of x86 and x64). When the extra xgcc toolchain is added, the default x86 and x64 targets added by gradle's internals cause gradle to think the platform is buildable, but when it looks for aarch64-linux-gnueabi-g++, it can't find it, and fails instantly.

### Steps to Reproduce (for bugs)

See above.

### Your Environment

Gradle Version: 4.1
Native Plugin: "cpp"

**Comment by JacisNonsense** [comment]
I've done a bit more digging and found a workable solution for the time being.

Subclassing `AbstractGccCompatibleToolChain` identically to the implementation of `GccToolChain`, and adding the following lines in the constructor will 'unset' the default x86 and x64 architectures added by the `AbstractGccCompatibleToolChain` constructor through reflection.
```groovy
def platConfField = AbstractGccCompatibleToolChain.class.getDeclaredField("platformConfigs")
platConfField.setAccessible(true)
platConfField.get(this).clear()
platConfField.setAccessible(false)

def configInsertLoc = AbstractGccCompatibleToolChain.class.getDeclaredField("configInsertLocation")
configInsertLoc.setAccessible(true)
configInsertLoc.set(this, 0)
configInsertLoc.setAccessible(false)
```

Perhaps it may be worthwhile to add a way to 'reset' the platformConfigs property of AbstractGccToolChain to make this behaviour accessible to the DSL?

**Comment by JacisNonsense** [comment]
Closed with #3124
